src/test/regress/sql/multi_alter_table_add_constraints_without_name.sql

```diff
@@ -0,0 +1,227 @@
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname = 'dist_partitioned_table';
+
+\c - - :public_worker_1_host :worker_1_port
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'longlonglonglonglonglonglonglonglong%' ORDER BY con.conname ASC;
+
+\c - - :master_host :master_port
+ALTER TABLE dist_partitioned_table DROP CONSTRAINT dist_partitioned_table_pkey;
+
+\c - - :public_worker_1_host :worker_1_port
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'longlonglonglonglonglonglonglonglong%' ORDER BY con.conname ASC;
+
+-- Test we error out when creating a primary key on a partition table with a long name if we cannot
+-- switch to sequential execution
+\c - - :master_host :master_port
+BEGIN;
+	SELECT count(*) FROM dist_partitioned_table;
+	ALTER TABLE dist_partitioned_table ADD PRIMARY KEY(partition_col);
+ROLLBACK;
+-- try inside a sequential block
+BEGIN;
+	SET LOCAL citus.multi_shard_modify_mode TO 'sequential';
+	SELECT count(*) FROM dist_partitioned_table;
+	ALTER TABLE dist_partitioned_table ADD PRIMARY KEY(partition_col);
+	ROLLBACK;
+
+-- Test primary key name is generated by postgres for citus local table.
+\c - - :master_host :master_port
+SET client_min_messages to ERROR;
+SELECT 1 FROM master_add_node('localhost', :master_port, groupId => 0);
+RESET client_min_messages;
+
+CREATE TABLE citus_local_table(id int, other_column int);
+SELECT citus_add_local_table_to_metadata('citus_local_table');
+
+ALTER TABLE citus_local_table ADD PRIMARY KEY(id);
+
+-- Check the primary key is created for the local table and its shard.
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'citus_local_table%' ORDER BY con.conname ASC;
+
+SELECT create_distributed_table('citus_local_table','id');
+
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'citus_local_table%' ORDER BY con.conname ASC;
+
+\c - - :public_worker_1_host :worker_1_port
+SELECT con.conname
+    FROM pg_catalog.pg_constraint con
+      INNER JOIN pg_catalog.pg_class rel ON rel.oid = con.conrelid
+      INNER JOIN pg_catalog.pg_namespace nsp ON nsp.oid = connamespace
+          WHERE rel.relname LIKE 'citus_local_table%' ORDER BY con.conname ASC;
+
```

Comment:
Citus local tables have shown problems with self-deadlocks. Partitioned tables as well.

JUST TO BE SURE, can we add one more test with "partitioned citus local table"?